python/ray/tests/test_cancel.py

```diff
@@ -258,5 +258,27 @@ def remote_wait(sg):
         ray.get(inner, timeout=10)
 
 
+@pytest.mark.parametrize("use_force", [True, False])
+def test_recursive_cancel(ray_start_regular, use_force):
+    @ray.remote
+    def inner():
+        while True:
+            time.sleep(0.1)
+
+    @ray.remote(num_cpus=1)
+    def outer():
+
+        x = [inner.remote()]
+        print(x)
+        while True:
+            time.sleep(0.1)
+
+    outer_fut = outer.remote()
+    outer_val = ray.get(outer_fut)
+    ray.cancel(outer_fut)
+    with pytest.raises(valid_exceptions(use_force)):
```

Comment:
I'm not super clear on how this test works right now, but one suggestion is to test whether the inner task got canceled by launching more tasks that would require the resources held by the inner task. You can also use the same trick to make sure that the inner task has gotten scheduled.

Reply:
I reworked it to use resources--that is definitely the best way to do this :)

Reply:
Should I use custom resources?

Reply:
I think just using CPUs should work!